portal-api/src/v1/controllers/deal-status/update-facility-cover-start-dates.js

```diff
@@ -1,42 +1,56 @@
 const CONSTANTS = require('../../../constants');
+const { getNowAsEpoch } = require('../../helpers/date');
 const facilitiesController = require('../facilities.controller');
 
+/**
+ * Updates the cover start dates of facilities in a deal.
+ * @param {object} user - The user object representing the user performing the update.
+ * @param {object} deal - The deal object containing the facilities to be updated.
+ * @returns {object} - The modified deal object with updated facility cover start dates.
+ */
 const updateFacilityCoverStartDates = async (user, deal) => {
-  const modifiedDeal = deal;
+  try {
+    const modifiedDeal = { ...deal };
 
-  if (modifiedDeal.facilities.length) {
-    modifiedDeal.facilities.forEach(async (facilityId) => {
-      const facility = await facilitiesController.findOne(facilityId);
+    if (!modifiedDeal.facilities || !modifiedDeal.facilities?.length) {
+      console.error('No facilities found in deal %s', modifiedDeal._id);
+      return modifiedDeal;
+    }
 
+    for (const facilityId of modifiedDeal.facilities) {
```

Comment:
for (const x of y) is much better than an async foreach so definitely happen with this - can we use a .map and do this at the same time instead of do we need it to be sequential? Can imagine facilitiesController.update() might overwrite some deal details and not be able to be in parallel so if that's the case can ignore me

Reply:
Moving to `for(...)` was done to allow sequential execution of the facilities update to avoid building up of any race condition, using `map` will not be advisable due to its ability to simuletaneously amend all facilities at the same time as you have rightly mentioned.

Reply:
Okay great, thanks for confirming will resolve this comment!